runtime/doc/api.txt

```diff
@@ -1906,8 +1907,8 @@ nvim_buf_get_option({buffer}, {name})                  *nvim_buf_get_option()*
         Option value
 
 nvim_buf_set_option({buffer}, {name}, {value})         *nvim_buf_set_option()*
-    Sets a buffer option value. Passing `nil` as value deletes the option
-    (only works if there's a global fallback)
+    Sets a buffer option value. Passing `nil` as the argument for value deletes 
```

Comment:
The new wording is more confusing and unnecessarily verbose. "The argument for value" is redundant, because value is an argument.